internal/proxy/util.go

```diff
@@ -487,3 +487,34 @@ func fillFieldIDBySchema(columns []*schemapb.FieldData, schema *schemapb.Collect
 
 	return nil
 }
+
+func validateUsername(username string) error {
+	username = strings.TrimSpace(username)
+
+	if username == "" {
+		return errors.New("username should not be empty")
+	}
+
+	invalidMsg := "Invalid username: " + username + ". "
+	if int64(len(username)) > Params.ProxyCfg.MaxUsernameLength {
+		msg := invalidMsg + "The length of username must be less than " +
+			strconv.FormatInt(Params.ProxyCfg.MaxUsernameLength, 10) + " characters."
+		return errors.New(msg)
+	}
+
+	firstChar := username[0]
+	if !isAlpha(firstChar) {
```

Comment:
does it make sense to limit the first char for a DB username?

Reply:
yes, it cannot be, like, a number or special character for the first character of a username.